src/test/java/gov/usgs/wma/waterdata/timeseries/TimeSeriesControllerXmlIT.java

```diff
@@ -0,0 +1,130 @@
+	public void found1095Test() {
+		runCase("monitoring-locations", "USGS-07227448", "e6a4cc2de5bf437e83efe0107cf026ac",
+				"xml/e6a4cc2de5bf437e83efe0107cf026ac_1095.xml");
+		runCase("SAT", "USGS-07227448", "e6a4cc2de5bf437e83efe0107cf026ac",
+				"xml/e6a4cc2de5bf437e83efe0107cf026ac_1095.xml");
+	}
+
+	@Test
+	public void mediaTypeNotAcceptableTest() {
+		String baseUrl = "/collections/monitoring-locations/items/USGS-07227448/observations/statistical-time-series/e6a4cc2de5bf437e83efe0107cf026ac";
+		for (String contentType : contentNotAccepted) {
+			ResponseEntity<String> rtn = restTemplate.getForEntity(buildUrl(baseUrl, contentType), String.class);
+			assertThat(rtn.getStatusCode(), equalTo(HttpStatus.BAD_REQUEST));
+			assertEquals(ogcNotAcceptedPayload, rtn.getBody());
+		}
+	}
+
+	@Test
+	public void mediaTypeNotAcceptableSatTest() {
+		String baseUrl = "/collections/SAT/items/USGS-07227448/observations/statistical-time-series/e6a4cc2de5bf437e83efe0107cf026ac";
+		for (String contentType : contentNotAccepted) {
+			ResponseEntity<String> rtn = restTemplate.getForEntity(buildUrl(baseUrl, contentType), String.class);
+			assertThat(rtn.getStatusCode(), equalTo(HttpStatus.BAD_REQUEST));
+			assertEquals(ogcNotAcceptedPayload, rtn.getBody());
+		}
+	}
+
+	@Test
+	public void notFoundTest() {
+		String url = "/collections/monitoring-locations/items/USGS-12345678/observations/statistical-time-series/e6a4cc2de5bf437e83efe0107cf026ac";
+		ResponseEntity<String> rtn = restTemplate.getForEntity(buildUrl(url, "xml"), String.class);
+		assertThat(rtn.getStatusCode(), equalTo(HttpStatus.NOT_FOUND));
+		assertEquals(ogc404Payload, rtn.getBody());
+	}
+
+	@Test
+	public void notFoundInSatCollectionTest() {
+		String url = "/collections/monitoring-locations/items/USGS-12345678/observations/statistical-time-series/e6a4cc2de5bf437e83efe0107cf026ac";
+		ResponseEntity<String> rtn = restTemplate.getForEntity(buildUrl(url, "xml"), String.class);
+		assertThat(rtn.getStatusCode(), equalTo(HttpStatus.NOT_FOUND));
+		assertEquals(ogc404Payload, rtn.getBody());
+	}
+
+	@Test
+	public void noGeomTest() {
+		String url = "/collections/monitoring-locations/items/USGS-04028090/observations/statistical-time-series/e6a4cc2de5bf437e83efe0107cf026ac";
+		ResponseEntity<String> rtn = restTemplate.getForEntity(buildUrl(url, "xml"), String.class);
+		assertThat(rtn.getStatusCode(), equalTo(HttpStatus.NOT_FOUND));
+		assertEquals(ogc404Payload, rtn.getBody());
+	}
+
+	@Test
+	public void noGeomSatCollectionTest() {
+		String url = "/collections/SAT/items/USGS-04028090/observations/statistical-time-series/e6a4cc2de5bf437e83efe0107cf026ac";
+		ResponseEntity<String> rtn = restTemplate.getForEntity(buildUrl(url, "xml"), String.class);
+		assertThat(rtn.getStatusCode(), equalTo(HttpStatus.NOT_FOUND));
+		assertEquals(ogc404Payload, rtn.getBody());
+	}
+
+	private void runCase(String collectionId, String featureId, String tsid, String compareFile) {
```

Comment:
I like the runCase concept here but you might want to return a boolean result and check it in the originating unit test. Otherwise, all fails will be in the same place: This method's assertions.

Reply:
Junit does put in a trace, so you you which line runCase was called from.